argilla_plugins/datasets/hf_sync.py

```diff
@@ -0,0 +1,185 @@
+import dataclasses
+import os
+import time
+from threading import Thread
+from typing import Optional, Set
+
+from datasets import Dataset, load_dataset
+
+import argilla as rg
+from argilla.client.sdk.commons.errors import NotFoundApiError
+from argilla.server.commons.models import TaskType
+
+
+@dataclasses.dataclass
+class Config:
+
+    hf_source: Optional[str] = None
+    hf_target: Optional[str] = None
+    hf_token: Optional[str] = None
+    hf_split: str = "train"
+    hf_push_to_hub_frequency: int = 60
+
+    rg_task: TaskType = TaskType.text_classification
```

Comment:
Why do we default to `text_classification`?

Reply:
Not sure. Yes, maybe can be a required value just to force users to which kind of dataset is expecting to sync.